### PR TITLE
Video metadata 2

### PR DESCRIPTION
**Pull request by kovacshuni**
One needs to set the following etcd values in the cluster manually:
* `etcdctl set /ft/_credentials/brightcove_auth ...`
* `etcdctl set /ft/config/brightcove/account-id ...`
* `etcdctl set /ft/config/bertha/urls/video-metadata-mappings ...`

the values can be found in lastpass.
Provisioner has been updated to create them this automatically when creating a new cluster.

One has to register two endpoints of CoCo manually through the Brightcove API to get notifications from them. See [BN README](https://github.com/Financial-Times/brightcove-notifier#register-notify-endpoint-with-brightcove-notification-api).
* https://pub-prod-up.ft.com/notification/brightcove/metadata
* https://pub-prod-up.ft.com/notification/brightcove/content

### CODE REVIEW

File: brightcove-metadata-notifier@.service [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[Unit]
+Description=brightcove metadata notifier
+After=docker.service
```
Comment (tosan88) [inline brightcove-metadata-notifier@.service]: Can you please use `vulcan.service` like here? https://github.com/Financial-Times/up-service-files/blob/master/template/template%40.service#L5

File: brightcove-metadata-notifier@.service [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Wants=brightcove-metadata-notifier-sidekick@%i.service
+
+[Service]
+Environment="DOCKER_APP_VERSION=latest"
+TimeoutStartSec=0
+# Change killmode from "control-group" to "none" to let Docker remove work correctly.
+KillMode=none
+ExecStartPre=-/bin/bash -c '/usr/bin/docker kill "$(docker ps -q --filter=name=%p-%i_)" > /dev/null 2>&1'
+ExecStartPre=-/bin/bash -c '/usr/bin/docker rm "$(docker ps -q --filter=name=%p-%i_)" > /dev/null 2>&1'
+ExecStartPre=/bin/bash -c 'docker pull coco/brightcove-metadata-notifier:$DOCKER_APP_VERSION'
```
Comment (tosan88) [inline brightcove-metadata-notifier@.service]: Can you please pull the image only if the image didn't get pulled before? https://github.com/Financial-Times/up-service-files/blob/master/template/template%40.service#L22

File: brightcove-notifier-sidekick@.service [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  etcdctl set   /ft/services/$SERVICE/healthcheck     true; \
+  etcdctl mkdir /ft/services/$SERVICE/servers; \
+  etcdctl set /ft/healthcheck/$SERVICE-%i/path /__health; \
+  while [ -z $PORT ]; do \
+    sleep 5; \
+    CONTAINER_NAME=$(docker ps -q --filter=name=\"$SERVICE\"-%i_); \
+    PORT=`echo $(/usr/bin/docker port $CONTAINER_NAME 8080 | cut -d':' -f2)`; \
+  done; \
+  etcdctl set /ft/services/$SERVICE/servers/%i http://%H:$PORT;"
+ExecStop=-/bin/bash -c 'etcdctl rm /vulcand/backends/brightcove-notifier/servers/%i'
```
Comment (tosan88) [inline brightcove-notifier-sidekick@.service]: I think you should remove `/ft/services/brightcove-notifier/servers/%i`

File: brightcove-notifier@.service [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+TimeoutStartSec=0
+# Change killmode from "control-group" to "none" to let Docker remove work correctly.
+KillMode=none
+ExecStartPre=-/bin/bash -c '/usr/bin/docker kill "$(docker ps -q --filter=name=%p-%i_)" > /dev/null 2>&1'
+ExecStartPre=-/bin/bash -c '/usr/bin/docker rm "$(docker ps -q --filter=name=%p-%i_)" > /dev/null 2>&1'
+ExecStartPre=/bin/bash -c '/usr/bin/docker history coco/brightcove-notifier:$DOCKER_APP_VERSION > /dev/null 2>&1 || /usr/bin/docker pull coco/brightcove-notifier:$DOCKER_APP_VERSION'
+
+ExecStart=/bin/sh -c '\
+  export APP_PORT=8080; \
+  export ENV=$(/usr/bin/etcdctl get /ft/config/environment_tag); \
```
Comment (tosan88) [inline brightcove-notifier@.service]: Unused variable

File: brightcove-metadata-notifier@.service (hunk 1 above)
Comment (kovacshuni) [inline brightcove-metadata-notifier@.service]: done.

File: brightcove-metadata-notifier@.service (hunk 2 above)
Comment (kovacshuni) [inline brightcove-metadata-notifier@.service]: done

File: brightcove-metadata-preprocessor@.service [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+TimeoutStartSec=0
+# Change killmode from "control-group" to "none" to let Docker remove work correctly.
+KillMode=none
+ExecStartPre=-/bin/bash -c '/usr/bin/docker kill "$(docker ps -q --filter=name=%p-%i_)" > /dev/null 2>&1'
+ExecStartPre=-/bin/bash -c '/usr/bin/docker rm "$(docker ps -q --filter=name=%p-%i_)" > /dev/null 2>&1'
+ExecStartPre=/bin/bash -c '/usr/bin/docker history coco/brightcove-notifier:$DOCKER_APP_VERSION > /dev/null 2>&1 || /usr/bin/docker pull coco/brightcove-notifier:$DOCKER_APP_VERSION'
+
+ExecStart=/bin/sh -c '\
+  export APP_PORT=8080; \
+  export ENV=$(/usr/bin/etcdctl get /ft/config/environment_tag); \
```
Comment (kovacshuni) [inline brightcove-metadata-preprocessor@.service]: removed.

File: brightcove-notifier@.service (hunk 4 above)
Comment (kovacshuni) [inline brightcove-notifier@.service]: removed.

File: brightcove-notifier-sidekick@.service (hunk 3 above)
Comment (kovacshuni) [inline brightcove-notifier-sidekick@.service]: corrected.

**Comment by tosan88** [comment]
:+1:

**Pull request merged**